lib/matplotlib/axes/_axes.py

```diff
@@ -1093,21 +1093,35 @@ def hlines(self, y, xmin, xmax, colors=None, linestyles='solid',
 
         lines = mcoll.LineCollection(masked_verts, colors=colors,
                                      linestyles=linestyles, label=label)
+        if 'transform' in kwargs:
+            lines.set_transform(kwargs['transform'])
         self.add_collection(lines, autolim=False)
+        self._unstale_viewLim()
         lines._internal_update(kwargs)
```

Comment:
So the last thing I'm left wondering is that your test passes without making these changes; are they still necessary now? Are they perhaps only needed for non-rectilinear Axes?

Reply:
I believe these are no longer necessary. I removed them.  Both the rectilinear and non-rectilinear tests in test_axes still pass without them.